cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -24,16 +42,33 @@ class ArgumentNode extends InstructionNode {
     instr = call.getPositionalArgument(pos)
     or
     instr = call.getThisArgument() and pos = -1
-    or
-    exists(ReadSideEffectInstruction read |
-      read = instr and
+  }
+}
+
+/**
+ * A data flow node representing the read side effect of a call on a
+ * specific parameter.
+ */
+private class SideEffectArgumentNode extends OperandNode {
+  override SideEffectOperand op;
+  ReadSideEffectInstruction read;
+
+  SideEffectArgumentNode() {
+    exists(CallInstruction call |
       read.getPrimaryInstruction() = call and
```

Comment:
What are these two lines for?

Reply:
That's a really good question! I'll remove them.

Reply:
Done in 6c87b08c69d10ecebc2889c5d4c1c4ffa09518f3.